### PR TITLE
Adding the RBAC List, Edit and Details pages

### PR DESCRIPTION
**Pull request by codyrancher**
rancher/dashboard#2111

Screenshot 2021-01-19 090123
Screenshot 2021-01-19 090022
Screenshot 2021-01-19 090008
Screenshot 2021-01-19 085858
Screenshot 2021-01-19 085838
Screenshot 2021-01-19 085818

### CODE REVIEW

File: models/rbac.authorization.k8s.io.roletemplate.js [hunk 1]
```diff
@@ -0,0 +1,228 @@
+import { uniq } from '@/utils/array';
+import Vue from 'vue';
+
+export const RESOURCES = [
```
Comment (vincent99) [inline models/rbac.authorization.k8s.io.roletemplate.js:4]: This list should come from schemas (`attributes.kind`), not be hardcoded into the UI.

File: edit/rbac.authorization.k8s.io.roletemplate.vue [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+              </div>
+            </div>
+          </template>
+          <template #columns="props">
+            <div class="columns row">
+              <div class="col span-3">
+                <Select
+                  v-model="props.row.value.verbs"
+                  class="lg"
+                  :taggable="false"
```
Comment (vincent99) [inline edit/rbac.authorization.k8s.io.roletemplate.vue:232]: Should be taggable; Other verbs exist and you should be able to enter them.

File: edit/rbac.authorization.k8s.io.roletemplate.vue [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+                  :searchable="true"
+                  :options="verbOptions"
+                  :multiple="true"
+                />
+              </div>
+              <div class="col span-3">
+                <Select
+                  :value="getRule('resources', props.row.value)"
+                  :options="resourceOptions"
```
Comment (vincent99) [inline edit/rbac.authorization.k8s.io.roletemplate.vue:241]: Should also be taggable

**Review by vincent99 — CHANGES_REQUESTED** [review]
Looks good but the dropdown fields should allow custom values

**Comment by lvuch** [comment]
this looks fine but i think we should reconsider the multi select dropdown app wide to be something more like this: 
image

**Review by lvuch — COMMENTED** [review]
review added above

File: models/rbac.authorization.k8s.io.roletemplate.js (hunk 1 above)
Comment (codyrancher) [reply models/rbac.authorization.k8s.io.roletemplate.js]: Done.

File: edit/rbac.authorization.k8s.io.roletemplate.vue (hunk 2 above)
Comment (codyrancher) [reply edit/rbac.authorization.k8s.io.roletemplate.vue]: Done

File: edit/rbac.authorization.k8s.io.roletemplate.vue (hunk 3 above)
Comment (codyrancher) [reply edit/rbac.authorization.k8s.io.roletemplate.vue:251]: Done

**Comment by codyrancher** [comment]
I created a new issue to track the LabeledSelect changes that we'd like to make here https://github.com/rancher/dashboard/issues/2277.

I ended up adding a 'Custom Verbs' column in the table so we could show when they were custom but we may want a better solution. Would you like to have a second table with custom verbs?
image

@vincent99 While working on filtering global, cluster and namespace resources I realized that my fields didn't match embers. Looking into a little further it seems like some of the ember fields aren't resources with schemas. This made me think my understanding may be a little off. An example of this is the for the project role there's a `binding` type which I couldn't find in v3/schemas. Is it fine that we just display resources we have?
image

**Comment by vincent99** [comment]
It looks like what ember does is list all other existing role templates and offer the resources which they use.  Which is sort of interesting, but circular, and lists a bunch of things that may not exist or are not known if they are namespaced or not...  I think what you have is fine.

**Pull request merged**